### PR TITLE
ESQL: Remove historical features

### PR DESCRIPTION
**Pull request by nik9000**
Our friends working on cluster features are looking to remove the infrastructure for historical features. In `main` all historical features are always enabled because historical features weren't allowed to support versions before 8.a_long_time_ago. All of ours are certainly enabled for versions `main` is wire compatible with.

### CODE REVIEW

File: x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plugin/TransportEsqlStatsAction.java [hunk 1]
```diff
@@ -63,13 +60,7 @@ public TransportEsqlStatsAction(
...
     @Override
     protected DiscoveryNode[] resolveRequest(EsqlStatsRequest request, ClusterState clusterState) {
-        if (featureService.clusterHasFeature(clusterState, ESQL_STATS_FEATURE)) {
-            // use the whole cluster
-            return super.resolveRequest(request, clusterState);
-        } else {
-            // not all nodes in the cluster have upgraded to esql - just use this node for now
-            return new DiscoveryNode[] { clusterService.localNode() };
-        }
+        return super.resolveRequest(request, clusterState);
```
Comment (thecoop) [inline x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plugin/TransportEsqlStatsAction.java:63]: nit - can just remove this override completely

**Review by thecoop — APPROVED** [review]
Thanks!

File: x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plugin/TransportEsqlStatsAction.java (hunk 1 above)
Comment (alex-spies) [inline x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plugin/TransportEsqlStatsAction.java:63]: Do we even need  the override if we defer to super?

**Review by alex-spies — APPROVED** [review]
LGTM in terms of ESQL, this should be safe to do.

File: x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/action/EsqlCapabilities.java [hunk 2]
```diff
@@ -140,6 +140,12 @@ public enum Cap {
          */
         CASE_MV,
 
+        /**
+         * Support for loading values over enrich. This is supported by all versions of ESQL but not
+         * the unit test CsvTests.
+         */
+        ENRICH_LOAD,
```
Comment (alex-spies) [inline x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/action/EsqlCapabilities.java:147]: nit: Maybe we want to turn the other features into caps as well just to keep number of divergent lines between 9.0 and 8.x minimal?

We can still remove old caps for good later.

File: x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plugin/TransportEsqlStatsAction.java (hunk 1 above)
Comment (nik9000) [reply x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plugin/TransportEsqlStatsAction.java:63]: Er, nope. sorry. Was being too quick on the refactor.

File: x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/action/EsqlCapabilities.java (hunk 2 above)
Comment (nik9000) [reply x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/action/EsqlCapabilities.java:147]: I *think* it's probably ok not to. It's going to be pretty rare that we make changes to the tests with these so I don't *think* the easier backports is going to come up.

**Pull request merged**